### PR TITLE
Week 4

### PR DESCRIPTION
**Pull request by Erkan-5Minds**

### CODE REVIEW

**Review by DevBastiS — CHANGES_REQUESTED** [review]
Remove ds store files

File: Week 4/filter-less/filter.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "helpers.h"
+
+int main(int argc, char *argv[])
+{
+    // Define allowable filters
```
Comment (DevBastiS) [inline Week 4/filter-less/filter.c:8]: Remove comments.

File: Week 4/filter-less/filter.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    int width = bi.biWidth;
+
+    // Allocate memory for image
+    RGBTRIPLE(*image)[width] = calloc(height, width * sizeof(RGBTRIPLE));
+    if (image == NULL)
+    {
+        printf("Not enough memory to store image.\n");
+        fclose(outptr);
+        fclose(inptr);
+        return 7;
```
Comment (DevBastiS) [inline Week 4/filter-less/filter.c:83]: Add a new line before returns.

File: Week 4/filter-more/helpers.c [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                image[i][j] = image[i][width - (j + 1)];
+                image[i][width - (j + 1)] = temp[i][j];
+            }
+        }
+    }
+
+    return;
+}
+
+// Blur image
```
Comment (DevBastiS) [inline Week 4/filter-more/helpers.c]: Remove comments

File: Week 4/recover/recover.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#include <stdint.h>
+typedef uint8_t BYTE;
+
+int main(int argc, char *argv[])
+{
+    if (argc != 2)
+    {
+        printf("Usage: .recover IMAGE\n");
+        return 1;
+    }
```
Comment (DevBastiS) [inline Week 4/recover/recover.c:12]:
```suggestion
    }
    
```

File: Week 4/recover/recover.c [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    FILE *input_file = fopen(argv[1], "r");
+    if (input_file == NULL)
+    {
+        printf("Could not open file");
+        return 2;
+    }
```
Comment (DevBastiS) [inline Week 4/recover/recover.c:19]:
```suggestion
    }
    
```

File: Week 4/recover/recover.c [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    char *filename = malloc(8 * sizeof(char));
+
+    while (fread (buffer, sizeof(char), 512, input_file))
+    {
+        if (buffer[0] == 0xff && buffer[1] == 0xd8 && buffer[2] == 0xff && (buffer[3] & 0xf0) == 0xe0)
+        {
+            sprintf(filename, "%03i.jpg", count_image);
+            output_file = fopen(filename, "w");
+            count_image++;
+        }
```
Comment (DevBastiS) [inline Week 4/recover/recover.c:33]:
```suggestion
        }
        
```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

**Review by stefanluth — COMMENTED** [review]
Remove `.DS_Store` & `.vscode/settings.json` :)

File: Week 4/recover/recover.c [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    unsigned char buffer[512];
+    int count_image = 0;
+    FILE *output_file = NULL;
+    char *filename = malloc(8 * sizeof(char));
+
+    while (fread (buffer, sizeof(char), 512, input_file))
+    {
+        if (buffer[0] == 0xff && buffer[1] == 0xd8 && buffer[2] == 0xff && (buffer[3] & 0xf0) == 0xe0)
+        {
+            sprintf(filename, "%03i.jpg", count_image);
```
Comment (stefanluth) [inline Week 4/recover/recover.c:30]: Does `sprintf()` exist or is this a typo? :)

**Review by DevBastiS — APPROVED** [review]

File: Week 4/recover/recover.c (hunk 7 above)
Comment (Erkan-5Minds) [reply Week 4/recover/recover.c:30]: its typo :)

**Review by stefanluth — APPROVED** [review]

**Pull request merged**